saleor/checkout/models.py

```diff
@@ -112,19 +117,17 @@ def get_subtotal(self, discounts=None):
 
     def get_total(self, discounts=None):
         """Return the total cost of the checkout."""
-        total = (
-            self.get_subtotal(discounts)
-            + self.get_shipping_price()
-            - self.discount_amount
-        )
+        total = self.get_subtotal(discounts) + self.get_shipping_price() - self.discount
```

Comment:
I think `get_subtotal` should use `zero_money(currency=self.currency)` instead of `DEFAULT_CURRENCY`.